feat: add --vaadin-app-layout-drawer-width CSS property

DiegoCardoso opened the pull request:
## Description

Adds the `--vaadin-app-layout-drawer-width` to be used as the value to be set as width to the drawer part of the App Layout. 

Fixes #950

## Type of change

- [ ] Bugfix
- [X] Feature

jouni commented:
The drawer offset should be overridden with the value of this new custom property value.

tomivirkki commented on packages/app-layout/src/vaadin-app-layout.js:
```diff
@@ -121,6 +121,7 @@ class AppLayout extends ElementMixin(ThemableMixin(ControllerMixin(PolymerElemen
           height: 100%;
           --vaadin-app-layout-transition: 200ms;
           transition: padding var(--vaadin-app-layout-transition);
+          --vaadin-app-layout-drawer-width: 16em;
```
There shouldn't be a default value for the `--vaadin-app-layout-drawer-width` property on the host because it would override any value applied to the property somewhere higher in the hierarchy.

For example, the following wouldn't work:

```css
body {
  --vaadin-app-layout-drawer-width: 150px;
}
```

DiegoCardoso replied:
Changed to define a private CSS prop that takes the `--vaadin-app-layout-drawer-width` and has a fallback value.

tomivirkki commented on packages/app-layout/src/vaadin-app-layout.js:
```diff
@@ -260,10 +261,7 @@ class AppLayout extends ElementMixin(ThemableMixin(ControllerMixin(PolymerElemen
-
-          [part='drawer'] {
-            width: 20em;
+            --vaadin-app-layout-drawer-width: 20em;
```
If you set:
```css
vaadin-app-layout {
  --vaadin-app-layout-drawer-width: 300px;
}
```

...it will be stronger than this value and the drawer width will have the same width on wide and narrow screen. Not sure what is the desired outcome. Should there be different properties for the two modes or should the default here be calculated from whatever the value is on the host by default?

jouni replied:
> Should there be different properties for the two modes

I don’t think so. Should be enough to scope them based on the mode attribute:

```css
vaadin-app-layout {
  --vaadin-app-layout-drawer-width: 100px;
}

vaadin-app-layout[overlay] {
  -vaadin-app-layout-drawer-width: 200px;
}
```

DiegoCardoso commented:
> The drawer offset ~should~ could be overridden with the value of this new custom property value, i.e., `--_vaadin-app-layout-drawer-offset-size: var(--vaadin-app-layout-drawer-width);`.
> 
> That said, there’s a risk the computed value could be different with `%` and `em` based values.

Changed to use the new prop and removed the bounding client rect calculation.
